src/components/routing/nested-routes.test.tsx

```diff
@@ -0,0 +1,113 @@
+import React from "react";
+import { NestedRoutes } from "./nested-routes";
+import { createMemoryHistory } from "history";
+import { render } from "@testing-library/react";
+import { Router } from "react-router-dom";
+import { RouteDefinition } from "../../interfaces/route-definition";
+import { Factory } from "rosie";
+import { FactoryType } from "../../tests/factories/factory-type";
+import faker from "faker";
+
+describe("NestedRoutes", () => {
+    let nonExistentRoute: string;
+    const HomePage = () => <h1>Home</h1>;
+    const NotFoundPage = () => <h1>404</h1>;
+
+    const homeRoute: RouteDefinition = Factory.build<RouteDefinition>(
+        FactoryType.RouteDefinition.Default,
+        { path: "/", component: HomePage }
+    );
+
+    const nestedRoutes: RouteDefinition[] = Factory.buildList(
+        FactoryType.RouteDefinition.Nested,
+        3
+    );
+
+    const notFoundRoute = Factory.build<RouteDefinition>(
+        FactoryType.RouteDefinition.Default,
+        { path: "/404", component: NotFoundPage }
+    );
+
+    let routes: RouteDefinition[] = [];
+
+    beforeEach(() => {
+        nonExistentRoute = [
+            "", // Preprends a / before the route
+            faker.random.alphaNumeric(5),
+            faker.random.alphaNumeric(5),
+            faker.random.alphaNumeric(5),
+        ].join("/");
+
+        routes = [...nestedRoutes, homeRoute, notFoundRoute];
+    });
+
+    // -----------------------------------------------------------------------------------------
+    // #region redirectToIfNotFound
+    // -----------------------------------------------------------------------------------------
+
+    describe("when pathname does not match any route path", () => {
+        describe("given redirectToIfNotFound has a value", () => {
+            test("it redirects to the redirectToIfNotFound route", () => {
+                // Arrange
+                const history = createMemoryHistory();
+                // This should have no effect on whether or not the user is redirected for a non-existent route
+                const isAuthenticated = faker.random.boolean();
+                const redirectToIfNotFound = notFoundRoute.path; // This is the important setup
+
+                const App = () => (
+                    <Router history={history}>
```

Comment:
Thought: Could be wrong, but think you could spare the extra setup of 'history' by using `MemoryRouter`.

Reply:
Ah, yeah, I was seeing that in the docs too. Wasn't sure how I would simulate a navigation change, though it looks like `MemoryRouter` has an `initialEntries` prop which may have worked the same way.